Add initial ServiceNow plugin docs (#31890)
gravitational/teleport: docs/pages/access-controls/access-request-plugins/servicenow.mdx

---
title: Access Requests with ServiceNow
description: How to set up Teleport's ServiceNow plugin for privilege elevation approvals.
---

With Teleport's ServiceNow integration, engineers can access the infrastructure
they need to resolve incidents quickly, without longstanding admin permissions
that can become a vector for attacks.

> Teleport's ServiceNow integration allows you to treat Teleport Role Access

Reviewer: Does it support resource access requests too, or only role requests?

Requests as ServiceNow incidents, notify the appropriate on-call team, and
approve or deny the requests via Teleport.

This guide will explain how to set up Teleport's Access Request plugin for
ServiceNow.

## Prerequisites

(!docs/pages/includes/commercial-prereqs-tabs.mdx!)

- An ServiceNow account with access to read and write to and from the 'incident' table.
- The ServiceNow integration is currently only available in Teleport Cloud.
- (!docs/pages/includes/tctl.mdx!)

## Step 1/4. Create services

We will configure the ServiceNow plugin to create an incident when
certain users create an Access Request.

## Step 2/4. Define RBAC resources

The Teleport ServiceNow plugin works by receiving Access Request events from the
Teleport Auth Service and, based on these events, interacting with the ServiceNow
API.

### Create a requester role

> To create a user first navigate to Management -> Access -> Roles

Reviewer: Don't we have a preset requester role now? If so, I think we should rewrite this to use the preset, instead of creating another one.

I would also remove the screenshot, the text description of the YAML is enough.

Author: What about changing it like this so it doesn't overwrite the preset role and creates one with less scope for this demo?

Reviewer: This is okay, but in the past Sasha has asked us to use the preset roles wherever possible to make the getting started experience as simple as we can.

If we already have a preset that includes the appropriate permissions, wouldn't it be much simpler to just use it?

Reviewer 2: @EdwardDowling Zac has a good point: we already have a builtin `requester` role that allows users to request `access` role. Let's just use it here.

plugin:

To create a user first navigate to Management -> Access -> Users

![Add user one](../../../img/enterprise/plugins/servicenow/add-user-one.png)

Then select 'Create New User' and create a user with the requester role.

![Add user two](../../../img/enterprise/plugins/servicenow/add-user-two.png)

## Step 3/4. Configure the ServiceNow plugin

At this point, you have a ServiceNow user that the ServiceNow plugin will use
to connect to the ServiceNow API. To configure the plugin to use this user navigate
to Management -> Integrations -> Enroll New Integration.

![Integrations page](../../../img/enterprise/plugins/servicenow/servicenow-integration-page.png)

For the plugin to function the credentials given to the plugin to use
must allow for reading and writing from the incident table.

## Step 4/4. Test the ServiceNow plugin

### Create an Access Request

As the Teleport user `myuser`, create an Access Request for the `editor` role:

(!docs/pages/includes/plugins/create-request.mdx!)

In ServiceNow, you will see a new incident containing information about the
Access Request.

### Resolve the request

(!docs/pages/includes/plugins/resolve-request.mdx!)

> <Admonition title="Auditing Access Requests">

Reviewer: I think it's odd for the last paragraph of the page to be an admonition. Is this necessary or can we just remove it?

---
title: Access Requests with ServiceNow
description: How to set up Teleport's ServiceNow plugin for privilege elevation approvals.
---

With Teleport's ServiceNow integration, engineers can access the infrastructure

> they need to resolve incidents quickly, without longstanding admin permissions

Reviewer: ```suggestion
they need to resolve incidents quickly, without granting longstanding admin permissions
```

To create a user first navigate to Management -> Access -> Roles

Then select 'Create New Role' and create the demo-requester role.

```
kind: role
version: v5
metadata:
  name: demo-requester
spec:
  allow:
    request:
      roles: ['editor']
      thresholds:
        - approve: 1
          deny: 1
```

### Create a user who will request access

Create a user called `myuser` who has the `demo-requester` role. Later in this
guide, you will create an Access Request as this user to test the ServiceNow
plugin:

To create a user first navigate to Management -> Access -> Users

![Add user one](../../../img/enterprise/plugins/servicenow/add-user-one.png)

Then select 'Create New User' and create a user with the demo-requester role.

![Add user two](../../../img/enterprise/plugins/servicenow/add-user-two.png)

## Step 3/4. Configure the ServiceNow plugin

At this point, you have a ServiceNow user that the ServiceNow plugin will use
to connect to the ServiceNow API. To configure the plugin to use this user navigate
to Management -> Integrations -> Enroll New Integration.

![Integrations page](../../../img/enterprise/plugins/servicenow/servicenow-integration-page.png)

> For the plugin to function the credentials given to the plugin to use
> must allow for reading and writing from the incident table.

Reviewer: We say "the plugin" twice here. What do you think about this wording?

```suggestion
The plugin requires credentials that can read and write to the incident table.
```
